pylops_mpi/utils/_nccl.py

```diff
@@ -251,61 +253,109 @@ def nccl_bcast(nccl_comm, local_array, index, value) -> None:
     )
 
 
-def nccl_asarray(nccl_comm, local_array, local_shapes, axis) -> cp.ndarray:
-    """Global view of the array
+def _prepare_nccl_allgather_inputs(send_buf, send_buf_shapes) -> tuple[cp.ndarray, cp.ndarray]:
+    """ Preparing the send_buf and recv_buf for the NCCL allgather (nccl_allgather)
 
-    Gather all local GPU arrays into a single global array via NCCL all-gather.
+    NCCL's allGather requires the sending buffer to have the same size for every device.
+    Therefore, the padding is required when the array is not evenly partitioned across
+    all the ranks. The padding is applied such that the sending buffer has the size of
+    each dimension corresponding to the max possible size of that dimension.
+
+    Receiver buff (recv_buf) will have the size n_rank * send_buf.size
```

Comment:
Similarly, each receiver buffer (`recv_buf`) is created with size equal to :math:`n_rank \cdot send_buf.size`